doc/feature/synapses.md

```diff
@@ -0,0 +1,189 @@
+    Synapses getAllSharedSynapses( const GIDSet& preGIDs,
+                                   const GIDSet& postGIDs ) const;
+};
+
+// read-only, moveable proxy object of data storage in Synapses container
+class Synapse
+{
+    uint32_t preGID() const;
+    uint32_t preSectionID() const;
+    uint32_t preSegmentID() const;
+    float preDistance() const;
+    Vector3f preSurfacePosition() const;
+    Vector3f preCenterPosition() const;
+
+    uint32_t postGID() const;
+    uint32_t postSectionID() const;
+    uint32_t postSegmentID() const;
+    float postDistance() const;
+    Vector3f postSurfacePosition() const;
+    Vector3f postCenterPosition() const;
+
+    float conductance() const;
+    float utilization() const;
+    float depression() const;
+    float facilitation() const;
+    float decay() const;
+    int efficacy() const;
+};
+
+// read-only, moveable, non-copyable, contains all data of all synapses
+class Synapses
+{
+    size_t size() const;
+    bool empty() const;
+
+    bool operator==( const Synapses& rhs ) const;
+    bool operator!=( const Synapses& rhs ) const;
+
+    const_iterator cbegin() const;
+    const_iterator cend() const;
+
+    const Synapse& operator[]( size_t ) const;
+
+    const uint32_t* preGID() const;
+    const uint32_t* preSectionID() const;
+    const uint32_t* preSegmentID() const;
+    const float* preDistance() const;
+    const float* preSurfacePositionX() const;
+    const float* preSurfacePositionY() const;
+    const float* preSurfacePositionZ() const;
+    const float* preCenterPositionX() const;
+    const float* preCenterPositionY() const;
+    const float* preCenterPositionZ() const;
+
+    const uint32_t* postGID() const;
+    const uint32_t* postSectionID() const;
+    const uint32_t* postSegmentID() const;
+    const float* postDistance() const;
+    const float* postSurfacePositionX() const;
+    const float* postSurfacePositionY() const;
+    const float* postSurfacePositionZ() const;
+    const float* postCenterPositionX() const;
+    const float* postCenterPositionY() const;
+    const float* postCenterPositionZ() const;
+
+    const float* conductance() const;
+    const float* utilization() const;
+    const float* depression() const;
+    const float* facilitation() const;
+    const float* decay() const;
+    const int* efficacy() const;
+
+    // intersection
+    Synapses operator&( const Synapses& rhs ) const;
+
+    // union
+    Synapses operator+( const Synapses& rhs ) const;
+    Synapses operator|( const Synapses& rhs ) const;
+
+    // disjunction (XOR)
+    Synapses operator^( const Synapses& rhs ) const;
+
+    // difference
+    Synapses operator-( const Synapses& rhs ) const;
+};
+
+// sequentially & forwarding iteration through synapses for any number of GIDs
+class SynapsesStream
+{
+    bool eos() const;
+
+    // async load of synapses for count cells
+    // throws if current + count leads to eos
+    std::future< Synapses > read( size_t count = 1 ) const;
```

Comment:
* w/o throw, there requested read is not complete. Could be an option as well, haven't much thought about both possibilities.
* No order guarantees implied
* Shared synapses need some more thinking, I have no concrete implementation even in mind
* The container can be empty, yes